libc/src/__support/str_to_float.h

```diff
@@ -1055,6 +1056,37 @@ hexadecimal_string_to_float(const char *__restrict src,
   return output;
 }
 
+struct NanMantissaResult {
```

Comment:
I don't think you need to define a separate struct for this.  You can use `cpp::expected` in `src/__support/CPP/expected.h`:
```
  using NanMantissaResult = cpp::expected<uint64_t, int>;
```